packages/paste-core/components/menu/src/index.tsx

```diff
@@ -1,12 +1,33 @@
-export * from "./useMenuState";
-export * from "./Menu";
-export * from "./MenuButton";
-export * from "./MenuBadge";
-export * from "./MenuGroup";
-export * from "./MenuItem";
-export * from "./MenuItem.styles";
-export * from "./MenuItemCheckbox";
-export * from "./MenuItemRadio";
-export * from "./MenuSeparator";
-export * from "./SubMenuButton";
-export * from "./types";
+import type { MenuInitialState, MenuStateReturn } from "./useMenuState";
+
+// for the sake of documenting the types we rename the state hook types to append Props to the name, so we can docuemnt them and not cause a breaking change.
+type MenuInitialStateProps = MenuInitialState;
+type MenuStateReturnProps = MenuStateReturn;
+
+export type { MenuInitialStateProps, MenuStateReturnProps };
```

Comment:
praise: I kinda liked how some of these packages with hooks seemed to follow a pattern of hook docs first followed by components. Was that intentional?

Reply:
just following the pattern from your first types PR of keeping them at the top of the file